lib/redirect.js

```diff
@@ -101,7 +101,12 @@ Redirect.prototype.onResponse = function (response) {
   }
 
   var uriPrev = request.uri
-  request.uri = url.parse(redirectTo)
+  
+  try{
+	  request.uri = url.parse(redirectTo)
+  } catch(e) {
+	  return false;
```

Comment:
If it's an invalid URL, it's an invalid URL. This should be an error that gets propagated to the callback via `request.emit('error', ...)` similar to https://github.com/request/request/pull/1873/files#diff-ecf293ae0a8a5d19f25e88fc9e60363aR94.